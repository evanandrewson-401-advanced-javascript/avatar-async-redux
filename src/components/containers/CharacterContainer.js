import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Characters from '../characters/Characters';
import { getCharacters, getCharacterLoading } from '../../selectors/charactersSelectors';
import { updateCharacters } from '../../actions/characterActions';

const CharacterContainer = () => {
  const characters = useSelector(getCharacters);
  const characterLoading = useSelector(getCharacterLoading);
  const dispatch = useDispatch();
  const updateCharacterList = () => dispatch(updateCharacters());

  useEffect(() => {
    updateCharacterList();
  }, []);

  if(characterLoading) return <h1>Loading...</h1>;

  return <Characters items={characters} />;
};

export default CharacterContainer;
