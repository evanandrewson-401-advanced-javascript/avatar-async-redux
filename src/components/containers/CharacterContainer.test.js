import React from 'react';
import { shallow } from 'enzyme';
import CharacterContainer from './CharacterContainer';
import store from '../../store';

// const characters = [{
//   _id: '5cf5679a915ecad153ab68c8',
//   allies: [
//     'Royal Earthbender Guards'
//   ],
//   enemies: [
//     'Chin'
//   ],
//   photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
//   name: '46th Earth King',
//   affiliation: ' Earth Kingdom Earth Kingdom Royal Family',
// }];

describe('CharacterContainer component', () => {
  it('renders CharacterContainer', () => {
    React.useContext = () => ({ store });
    const wrapper = shallow(<CharacterContainer />, {
      context: {
        store
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
