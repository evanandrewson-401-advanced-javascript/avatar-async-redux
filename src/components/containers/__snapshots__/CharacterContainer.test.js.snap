// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer component renders CharacterContainer 1`] = `
<Characters
  items={Array []}
/>
`;
